public/app/core/services/echo/backends/analytics/RudderstackBackend.ts

```diff
@@ -0,0 +1,74 @@
+import $ from 'jquery';
+import { EchoBackend, EchoEventType, isInteractionEvent, isPageviewEvent, PageviewEchoEvent } from '@grafana/runtime';
+import { User } from '../sentry/types';
+
+export interface RudderstackBackendOptions {
+  writeKey: string;
+  dataPlaneUrl: string;
+  user?: User;
+}
+
+export class RudderstackBackend implements EchoBackend<PageviewEchoEvent, RudderstackBackendOptions> {
+  supportedEvents = [EchoEventType.Pageview, EchoEventType.Interaction];
+
+  constructor(public options: RudderstackBackendOptions) {
+    const url = `https://cdn.rudderlabs.com/v1/rudder-analytics.min.js`;
+
+    $.ajax({
+      url,
+      dataType: 'script',
+      cache: true,
+    });
+
+    const rds = ((window as any).rudderanalytics = []);
+
+    var methods = [
+      'load',
+      'page',
+      'track',
+      'identify',
+      'alias',
+      'group',
+      'ready',
+      'reset',
+      'getAnonymousId',
+      'setAnonymousId',
+    ];
+
+    for (let i = 0; i < methods.length; i++) {
+      const method = methods[i];
+      (rds as Record<string, any>)[method] = (function (methodName) {
+        return function () {
+          // @ts-ignore
+          rds.push([methodName].concat(Array.prototype.slice.call(arguments)));
+        };
+      })(method);
+    }
+
+    (rds as any).load(options.writeKey, options.dataPlaneUrl);
+
+    if (options.user) {
```

Comment:
Do we have a way of adding a user if you have logged in in this session?

Reply:
If I start as a unauthenticated user and then login. Do we re-create the backend or can we add the user information after login?

Reply:
Technicaly not. But that's not required as there is a full page reload happening after the signing in, so the app init is performed again.

Reply:
Good question, in the old days the entire page would reload at login, afaik that's still true for oauth/saml login due to the redirect flow but I'm not sure if it's still the case for u/p login